servers/SR430 Scalar Server.py

```diff
@@ -0,0 +1,235 @@
+
+    @setting(12, 'Discriminator Level', value='v[V]', returns='s')
+    def discriminator_level(self, c, value=None):
+        '''
+        Sets or queries the discriminator level.
+         Acceptable range: [-0.300,0.300]
+         Equivalent to DCLV.
+        '''
+        dev = self.selectedDevice(c)
+        if value==None:
+            yield dev.write('DCLV?')
+            sleep(0.1)
+            message = yield dev.read()
+            returnValue(message)
+        else:
+            yield dev.write('DCLV '+str(value['V']))
+
+    @setting(13, 'Records Per Scan', value='w', returns='s')
+    def records_per_scan(self, c, value=None):
+        '''
+        Sets or queries the records per scan
+         Acceptable range: [0,65535]
+         Equivalent to RSCN.
+        '''
+        dev = self.selectedDevice(c)
+        if value==None:
+            yield dev.write('RSCN?')
+            sleep(0.1)
+            message = yield dev.read()
+            returnValue(message)
+        elif value > 65535:
+            message = "Input out of range.  Acceptable range: [0,65535]
+            returnValue(message)
+        else:
+            yield dev.write('RSCN '+str(value))
+
+    @setting(14, 'Output GPIB')
+    def output_gpib(self, c):
+        '''
+        Sets query output to GPIB
+         
+         Equivlanet to OUTP 1.
+        '''
+        dev = self.selectedDevice(c)
+        yield dev.write('OUTP 1')
+
+    @setting(15, 'Start Scan')
+    def start_scan(self, c):
+        '''
+        Starts scan (without clearing).  Same as pressing the [START] key.
+         For starting a scan with clearing, use start_new_scan().
+
+         Equivalent to SSCN.
+        '''
+        dev = self.selectedDevice(c)
+        yield dev.write('SSCN')
+
+    @setting(16, 'Stop Scan')
+    def stop_scan(self, c):
+        '''
+        Pauses a scan in progress.  Same as pressing [STOP] key.
+
+         Equivalent to PAUS.
+        '''
+        dev = self.selectedDevice(c)
+        yield dev.write('PAUS')
+
+    @setting(17, 'Clear Scan')
+    def clear_scan(self, c):
+        '''
+        Resets the unit to CLEAR state, losing all data.  Same as pressing [STOP],[STOP]
+
+         Equivalent to CLRS
+        '''
+        dev = self.selectedDevice(c)
+        yield dev.write('CLRS')
+
+    @setting(18, 'Start New Scan')
+    def start_new_scan(self, c):
+        '''
+        Clears current data and starts new scan.
+         Uses both clear_scan() and start_scan() methods.
+        '''
+        dev = self.selectedDevice(c)
+        yield self.clear_scan(c)
+        sleep(0.01)
+        yield self.start_scan(c)
+
+    @setting(19, 'Statistics')
+    def statistics(self, c):
+        '''
+        Performs statistics.
+
+         Equivalent to STAT
+        '''
+        dev = self.selectedDevice(c)
+        yield dev.write('STAT')
+
+    @setting(20, 'Bins Per Record', value='w', returns='s')
```

Comment:
We need to make it clear what the argument means. We should make a list of the allowed values and let the user choose one from list only. See martinis serial server baudrate function for an example.